docs/index.rst

```diff
@@ -151,6 +151,31 @@ on a CI/build server).
 * ``TWINE_NON_INTERACTIVE`` - Do not interactively prompt for username/password
   if the required credentials are missing.
 
+Proxy Support
+^^^^^^^^^^^^^
+
+Twine can be configured to use a proxy by setting environment variables.
+For example:
+
+.. code-block:: bash
+
+   export HTTPS_PROXY=socks5://user:pass@host:port
+
+   twine upload dist/*
+
+Alternatively, one can set the environment variable for the command
+itself without exporting it for other tools as well:
+
+.. code-block:: bash
+
+   HTTPS_PROXY=socks5://user:pass@host:port twine upload dist/*
```

Comment:
Okay, definitely bike-shedding here, but I'd prefer to keep this simple, rather than getting into the theory and practice of environment variables:

```suggestion
Twine can be configured to use a proxy by setting environment variables.
For example, to use a proxy for just the ``twine`` command,
without ``export``-ing it for other tools:

.. code-block:: bash

   HTTPS_PROXY=socks5://user:pass@host:port twine upload dist/*
```